If a key doesn't exists updatecli crashes

**Describe the bug**
Trying to check a key which doesn't exists in a condition results in the crash of updatecli.

**Steps to Reproduce**
For @dduportal:
https://github.com/jenkins-infra/kubernetes-management/pull/1881
In this PR I've trimmed [the podTemplate](https://github.com/jenkins-infra/kubernetes-management/blob/25fd5a8d4478b371c0e3b372247a890c35bd9d5c/PodTemplates.yaml) to keep only one container template, thus [this manifest](https://github.com/jenkins-infra/kubernetes-management/blob/7d62634660e13ec52171d70aedca316159b7becd/updatecli/updatecli.d/pod-templates/updatecli.yaml#L27) (and the yamllint too) references a key which doesn't exist anymore.

<details><summary>error log:</summary>

```

##############################

# BUMP UPDATECLI PODTEMPLATE #

##############################



SOURCES

=======


lastRelease

-----------

Searching for version matching pattern "latest"

✔ Github Release version "v0.17.2" found matching pattern "latest"



CHANGELOG:

----------


Release published on the 2022-01-03 09:30:04 +0000 UTC at the url https://github.com/updatecli/updatecli/releases/tag/v0.17.2


## Changes


## 🐛 Bug Fixes


- Bugfix: Don't try to open a pullrequest when not needed @olblak (#436)


## 🧰 Maintenance


- fix: correct regexp to retrieve last (non beta) go version @lemeurherve (#429)


## Contributors


@dduportal, @lemeurherve and @olblak




CONDITIONS:

===========


checkPodTemplate

----------------

panic: runtime error: index out of range [1] with length 1


goroutine 1 [running]:

github.com/updatecli/updatecli/pkg/plugins/yaml.replace(0xc0007474a0, {0xc00107a040, 0x2, 0x2}, {0xc000ddc740, 0x9}, 0x3)

	/home/runner/work/updatecli/updatecli/pkg/plugins/yaml/main.go:223 +0xa5c

github.com/updatecli/updatecli/pkg/plugins/yaml.replace(0xc000746be0, {0xc00107a030, 0x3, 0x3}, {0xc000ddc740, 0x9}, 0x1)

	/home/runner/work/updatecli/updatecli/pkg/plugins/yaml/main.go:274 +0x838

github.com/updatecli/updatecli/pkg/plugins/yaml.replace(0xc000746aa0, {0xc00107a030, 0x3, 0x3}, {0xc000ddc740, 0x9}, 0x1)

	/home/runner/work/updatecli/updatecli/pkg/plugins/yaml/main.go:274 +0x838

github.com/updatecli/updatecli/pkg/plugins/yaml.(*Yaml).condition(0xc0009b6ba0, {0x0, 0x0})

	/home/runner/work/updatecli/updatecli/pkg/plugins/yaml/condition.go:53 +0x1dd

github.com/updatecli/updatecli/pkg/plugins/yaml.(*Yaml).Condition(0xc0006d3878, {0x0, 0x40d047})

	/home/runner/work/updatecli/updatecli/pkg/plugins/yaml/condition.go:16 +0x1e

github.com/updatecli/updatecli/pkg/core/pipeline/condition.(*Condition).Run(0xc0006d3878, {0x0, 0x0})

	/home/runner/work/updatecli/updatecli/pkg/core/pipeline/condition/main.go:108 +0x4bc

github.com/updatecli/updatecli/pkg/core/pipeline.(*Pipeline).RunConditions(0xc000a5c000)

	/home/runner/work/updatecli/updatecli/pkg/core/pipeline/conditions.go:40 +0x49c

github.com/updatecli/updatecli/pkg/core/pipeline.(*Pipeline).Run(0xc000a5c000)

	/home/runner/work/updatecli/updatecli/pkg/core/pipeline/main.go:210 +0x225

github.com/updatecli/updatecli/pkg/core/engine.(*Engine).Run(0x3522ec0)

	/home/runner/work/updatecli/updatecli/pkg/core/engine/main.go:180 +0x594

github.com/updatecli/updatecli/cmd.run({0x1f54312, 0xc00013a500})

	/home/runner/work/updatecli/updatecli/cmd/root.go:100 +0x171

github.com/updatecli/updatecli/cmd.glob..func2(0x3505900, {0x1f54412, 0x4, 0x4})

	/home/runner/work/updatecli/updatecli/cmd/diff.go:27 +0xfd

github.com/spf13/cobra.(*Command).execute(0x3505900, {0xc00019c100, 0x4, 0x4})

	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:860 +0x5f8

github.com/spf13/cobra.(*Command).ExecuteC(0x3505e00)

	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:974 +0x3bc

github.com/spf13/cobra.(*Command).Execute(...)

	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:902

github.com/updatecli/updatecli/cmd.Execute()

	/home/runner/work/updatecli/updatecli/cmd/root.go:40 +0x65

main.main()

	/home/runner/work/updatecli/updatecli/main.go:8 +0x17

script returned exit code 2

```

</details>

**Expected behavior**
There should be an error message, not a crash.

**Screenshots**
<!-- If applicable, add screenshots to help explain your problem -->
N/A

**Version**
<!-- Please complete the following information -->
- OS: Alpine image (jenkins/inbound-agent:4.11-1-alpine-jdk11)
- Updatecli version: v0.17.2

**Additional context**
N/A

--- a/pkg/plugins/yaml/main.go
+++ b/pkg/plugins/yaml/main.go
@@ -218,7 +218,7 @@ func replace(entry *yaml.Node, keys []string, version string, columnRef int) (fo
 
 					positionIndex := position
 
-					if positionIndex > len(entry.Content[index+1].Content) {
+					if positionIndex >= len(entry.Content[index+1].Content) {
 						return false, "", 0
 					}
 

--- a/pkg/plugins/yaml/main_test.go
+++ b/pkg/plugins/yaml/main_test.go
@@ -5,6 +5,7 @@ import (
 	"testing"
 
 	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 	"github.com/updatecli/updatecli/pkg/core/text"
 	"gopkg.in/yaml.v3"
@@ -211,7 +212,7 @@ func TestReplace(t *testing.T) {
 
 	//https://github.com/go-yaml/yaml/issues/599
 
-	dataset := []dataSet{
+	dataset1 := []dataSet{
 		{
 			key:                []string{"image", "tag"},
 			expectedOldVersion: "1.17",
@@ -246,6 +247,11 @@ func TestReplace(t *testing.T) {
 			key:                []string{"image3[3]", "version"},
 			expectedOldVersion: "5.5",
 			expectedValueFound: true,
+		},
+		{
+			key:                []string{"image3[4]", "version"},
+			expectedOldVersion: "",
+			expectedValueFound: false,
 		},
 		{
 			key:                []string{"image5::tag"},
@@ -310,38 +316,24 @@ func TestReplace(t *testing.T) {
 	out := yaml.Node{}
 
 	err := yaml.Unmarshal([]byte(data), &out)
-	if err != nil {
-		logrus.Errorf("err - %s", err)
-	}
+	require.NoError(t, err)
 
-	for _, d := range dataset {
+	for _, d := range dataset1 {
 		valueFound, oldVersion, _ := replace(&out, d.key, source, 1)
 
-		if valueFound != d.expectedValueFound {
-			t.Errorf("Value not found for key %v! expected %v, got %v", d.key, d.expectedValueFound, valueFound)
-		}
-
-		if oldVersion != d.expectedOldVersion {
-			t.Errorf("Old Version mismatch for key %v! expected %v, got %v", d.key, d.expectedOldVersion, oldVersion)
-		}
+		assert.Equal(t, d.expectedValueFound, valueFound)
+		assert.Equal(t, d.expectedOldVersion, oldVersion)
 	}
 
 	out2 := yaml.Node{}
 	err = yaml.Unmarshal([]byte(data2), &out2)
-	if err != nil {
-		logrus.Errorf("err - %s", err)
-	}
+	require.NoError(t, err)
 
 	for _, d := range dataset2 {
 		valueFound, oldVersion, _ := replace(&out2, d.key, source, 1)
 
-		if valueFound != d.expectedValueFound {
-			t.Errorf("Value not found for key %v! expected %v, got %v", d.key, d.expectedValueFound, valueFound)
-		}
-
-		if oldVersion != d.expectedOldVersion {
-			t.Errorf("Old Version mismatch for key %v! expected %v, got %v", d.key, d.expectedOldVersion, oldVersion)
-		}
+		assert.Equal(t, d.expectedValueFound, valueFound)
+		assert.Equal(t, d.expectedOldVersion, oldVersion)
 	}
 }
 

--- a/pkg/plugins/yaml/test.yaml
+++ b/pkg/plugins/yaml/test.yaml
@@ -1,3 +1,0 @@
-github:
-  owner: obiwankenobi
-  repository: charts

